homework-solutions/js/hw-4/task-3.js

```diff
@@ -6,3 +6,31 @@
   преобразовываясь в number
 
 */
+
+let minAge = 18;
+let maxAge = 60;
+
+// Тестовые значения
+let testAges = [10, 17, 18, 19, 59, 60, 61, "18", "2", "abc", "20a", true, null, undefined];
+
+for (let input of testAges) {
+  console.log(`\nПроверка значения: ${input}`);
+
+  // Преобразуем строку, если она содержит только цифры
+  if (typeof input === "string" && /^\d+$/.test(input)) {
+    input = Number(input);
+  }
+
+  // Проверка типа данных после возможного преобразования
+  if (typeof input !== "number" || isNaN(input)) {
+    console.log("Incorrect data type");
+  } else if (input < minAge && input > 0) {
```

Comment:
Если я не ошибаюсь для данного задания достаточно проверки `isNaN` 
```suggestion

  if (isNaN(input)) {
    console.log("Incorrect data type");
   } else {
     input = Number(input) // If previous check passes then convert to number type
```